### PR TITLE
Fixed AudioPlayer giving IllegalStateException on stop

### LINKED ISSUE #11363: [Bug] There seems to be some issues with the recording function, which sometimes caused my phone to stuck
###### Reproduction Steps

1. Record audio
2. Click Record
3. Click to pause recording
4. Click Play
5. Click Stop
6. Click Play again
7. Click pause
8. All three buttons will no longer respond
9. Even after quitting this recording, both ankidroid and my phone became strange and I had to restart my phone

###### Expected Result

Won't get stuck and keep reporting errors

###### Actual Result

Stuck and kept reporting errors

###### Debug info

AnkiDroid Version = 2.16alpha63
Android Version = 11
Manufacturer = OnePlus
Model = KB2000
Hardware = qcom
Webview User Agent = Mozilla/5.0 (Linux; Android 11; KB2000 Build/RP1A.201005.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/99.0.4844.88 Mobile Safari/537.36
ACRA UUID = aa4d19d2-311f-41e8-87a9-bc4ce5082d08
Scheduler = std2
Crash Reports Enabled = true
DatabaseV2 Enabled = true

###### Video

https://user-images.githubusercontent.com/15941953/168757681-ce6946b3-2125-417c-a043-bb44b7567daf.mp4

###### Research
*Enter an [x] character to confirm the points below:*
- [x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x ] I have searched for similar existing issues here and on the user forum
- [x ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by dorrin-sot**
## Purpose / Description
Fixed bug by preventing the call to `MediaPlayer::prepare` from happening while in `STARTED` state

## Fixes
Fixes #11363 

## Approach
Based on [the native Android `AudioPlayer`'s State Diagram](https://developer.android.com/reference/android/media/MediaPlayer#state-diagram) `mPlayer.prepare` cannot be called while in `STARTED` state.

[Android MediaPlayer docs](https://developer.android.com/reference/android/media/MediaPlayer)
[MediaPlayer State Numbers](https://stackoverflow.com/a/29036185/12725405)

## How Has This Been Tested?
Emulator (Doesn't throw `IllegalStateException`)
Real Device (All the functions of start, stop, pause and resume working properly)

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by dorrin-sot** [comment]
> 🤔

Fixed but I had to add a whole new MediaPlayer class just so I can track state and find out if we are `start`ing after `pause`ing or `stop`ing

**Review by mikehardy — APPROVED** [review]
Oh wow, that's really obnoxious that the only way it seems to work is by duplicating the state machine in local / non-private code.

It does look like a valid solution though, nice

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/AudioPlayer.kt [hunk 1]
```diff
@@ -50,13 +50,15 @@ class AudioPlayer {
...
+            mPlayer!!.prepare()
+        }
         mPlayer!!.start()
     }
 
     fun stop() {
         try {
-            mPlayer!!.prepare()
-            mPlayer!!.seekTo(0)
+            mPlayer!!.stop()
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/AudioPlayer.kt:61]: This seems to be a functional change: we no longer `seekTo(0)`, is this intentional?

Comment (dorrin-sot) [reply AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/AudioPlayer.kt:61]: Yes `MediaPlayer.stop()` also seeks to start.

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/AudioPlayer.kt:61]: Ah, got it:

> If playback had been stopped, or never started before, playback will start at the beginning.

From the `start()` documentation

**Review by david-allison — APPROVED** [review]

**Pull request merged**